language/polkavm/move-to-polka/src/tools.rs

```diff
@@ -25,66 +30,83 @@ impl PlatformTools {
 
         Ok(())
     }
+
+    pub fn get_runtime(&self, out_path: &PathBuf) -> anyhow::Result<PathBuf> {
+        debug!("building move-native runtime for polkavm in {out_path:?}");
+        println!("building move-native runtime for polkavm in {out_path:?}");
+        let archive_file = out_path
+            .join("riscv32imac-unknown-none-elf")
+            .join("release")
+            .join("libmove_native.a");
+
+        if archive_file.exists() {
+            return Ok(archive_file);
+        }
+
+        let move_native = std::env::var("MOVE_NATIVE").expect("move native");
+        let move_native = PathBuf::from(move_native);
+        let move_native = move_native.join("Cargo.toml").to_string_lossy().to_string();
+
+        // Using `cargo rustc` to compile move-native as a staticlib.
+        // See move-native documentation on `no-std` compatibilty for explanation.
+        // Release mode is required to eliminate large stack frames.
+        let res = self.run_cargo(
+            out_path,
+            &[
+                "rustc",
+                "--crate-type=staticlib",
+                "-p",
+                "move-native",
+                "--target",
+                "riscv32imac-unknown-none-elf",
+                "--manifest-path",
+                &move_native,
+                "--release",
+                "--features",
+                "polkavm",
+                // "-q",
+            ],
+        );
+
+        if let Err(e) = res {
+            anyhow::bail!("{e}");
+        }
+
+        if !archive_file.exists() {
+            anyhow::bail!("native runtime not found at {archive_file:?}. this is a bug");
+        }
+
+        Ok(archive_file)
+    }
+
+    pub fn merge_object_files(&self, sources: &[PathBuf], output: &PathBuf) -> anyhow::Result<()> {
+        let output = Command::new(&self.lld)
+            .arg("-r")
+            .arg("-o")
+            .arg(output)
+            .args(sources)
+            .output()?;
+        let status = output.status;
+        if !status.success() {
+            error!("ld.ldd execution error:");
```

Comment:
Should this be `ld.lld` instead of `ld.ldd`?

Reply:
Ha ha. Changed real binary. Forgot to change error message. Good catch.